lazy_load_symbols/load_sym_lazily.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import absolute_import
+import re
+from subprocess import Popen, PIPE
+import gdb
+
+
+def parse_raw_symlist(syml):
+    symre = r' ([^ ]*)$'
+    symc = re.compile(symre)
+    pl = []
+    for line in syml.split('\n'):
+        m = symc.search(line)
+        if m:
+            pl.append(m.group(1))
+    return pl
+
+
+def extract_symbols_from_raw_list(rs):
+    '''
+    Saves the entire symbol names as a single long string
+    so we can say something like:
+        if symbol in symbol_list
+    '''
+    pl = ''
+    for sym in rs:
+        filth = Popen(['c++filt', sym], stdout=PIPE)
+        (sym_name, err) = filth.communicate()
+        assert not err, 'c++filt failed {}'.format(err)
+        assert sym_name, 'c++filt didn\'t demangle {}'.format(sym)
+        ec = filth.wait()
+        assert not ec, 'c++filt exited with {}'.format(ec)
+        pl += sym_name
+    return pl
+
+
+class LoadLazySymbols(gdb.Command):
+    '''
+    A command to load a recording without loading
+    the symbols and then loading only the required symbols.
+    '''
+
+    def __init__(self):
+        super(LoadLazySymbols, self).__init__('loadlazy', gdb.COMPLETE_EXPRESSION)
+        self.solist = []
+        self.sodict = {}
+
+    def load_so_list(self):
+        self.solist = gdb.execute('info sharedlibrary', to_string=True)
```

Comment:
Is the same information available somehow through the public Python API?
For instance, look at https://sourceware.org/gdb/onlinedocs/gdb/Objfiles-In-Python.html#Objfiles-In-Python and https://sourceware.org/gdb/onlinedocs/gdb/Progspaces-In-Python.html#Progspaces-In-Python.

Reply:
that is a very good point, I think that I might be able to simplify things dramatically using this interface